Detectors/Vertexing/src/SVertexer.cxx

```diff
@@ -580,11 +605,13 @@ bool SVertexer::checkV0(const TrackCand& seedP, const TrackCand& seedN, int iP,
   }
   // check tight lambda mass only
   bool goodLamForCascade = false, goodALamForCascade = false;
-  if (mV0Hyps[Lambda].checkTight(p2Pos, p2Neg, p2V0, ptV0)) {
-    goodLamForCascade = true;
-  }
-  if (mV0Hyps[AntiLambda].checkTight(p2Pos, p2Neg, p2V0, ptV0)) {
-    goodALamForCascade = true;
+  if (ptV0 > mSVParams->minPtV0FromCascade && seedP.nITSclu >= mSVParams->mITSSAminNcluCascades && seedN.nITSclu >= mSVParams->mITSSAminNcluCascades) {
```

Comment:
But now you are rejecting all tracks with short ITS segment, even if it had TPC. Should  not it be 
```
if (ptV0 > mSVParams->minPtV0FromCascade && ((seed.hasTPC && seedP.nITSclu>0 )|| (seedP.nITSclu >= mSVParams->mITSSAminNcluCascades && seedN.nITSclu >= mSVParams->mITSSAminNcluCascades)) {
```
?
Note that above I also excluded TPC-only tracks in case they are allowed (ping to @f3sch ).

I guess one should add such a check also for the bachelor, e.g. here https://github.com/AliceO2Group/AliceO2/blob/e927ff6df7c18418957cd3e1f23c0a5c327640aa/Detectors/Vertexing/src/SVertexer.cxx#L776
```
if (bach.nITSclu<1 || (!baxh.hasTPC && bach.nITSclu<mSVParams->mITSSAminNcluCascades) {
  continue;
}
```
In principle, instead of keeping hasTPC and nClu data-members of the track, we can have a single flag e.g. goodForCascade, set as `(hasTPC&&nClu>0) || nClu>=mSVParams->mITSSAminNcluCascades)`

Reply:
Hi Ruben, you're right, sorry. I changed it now to: 

```
  // check tight lambda mass only
  bool goodLamForCascade = false, goodALamForCascade = false;
  if( ptV0 > mSVParams->minPtV0FromCascade && (seedP.hasTPC || seedP.nITSclu >= mSVParams->mITSSAminNcluCascades) && (seedN.hasTPC || seedN.nITSclu >= mSVParams->mITSSAminNcluCascades) ){
    if (mV0Hyps[Lambda].checkTight(p2Pos, p2Neg, p2V0, ptV0) && (!mSVParams->mRequireTPCforCascBaryons || seedP.hasTPC) && seedP.compatibleProton) {
      goodLamForCascade = true;
    }
    if (mV0Hyps[AntiLambda].checkTight(p2Pos, p2Neg, p2V0, ptV0) && (!mSVParams->mRequireTPCforCascBaryons || seedN.hasTPC && seedN.compatibleProton)) {
      goodALamForCascade = true;
    }
  }
```
and then the bachelor indeed requires: 
```
    if( !bach.hasTPC && bach.nITSclu < mSVParams->mITSSAminNcluCascades ){ 
      continue; 
    }
```
I'll reply on JIRA too. Thanks!

Reply:
One more comment: I deliberately want to avoid a generic flag "is good for cascade", because in fact, it is at the V0 usage stage that we can decide on applying *specific* selections on baryons and mesons (because of the bachelor charge), so I would prefer to keep as is if possible. Thanks!

Reply:
OK, but still you allow TPC only tracks. @f3sch was going to exclude them but will happen on the same lines you edited, could do at once.

Reply:
Hi @shahor02 , so after talking to @f3sch I added a configurable rejection of TPC-only tracks for cascading and for the 3-body decays as well for the sake of the future. We also had a cool discussion on the problem of TPC-only tracks for PCMs :-) 

I've now moved this to ready for review. Please check what I did just in case I slipped up, but hopefully we should be good to go now! Thank you and have a nice weekend!